src/editor/Core/Ripgrep.re

```diff
@@ -0,0 +1,38 @@
+[@deriving show]
+type t = {search: string => list(string)};
+
+let mergeOptions = options =>
+  List.fold_left((accum, opt) => accum ++ " " ++ opt, "", options);
+
+let process = (rgPath, args) => {
+  let lines = ref([]);
+  /**
+     NOTE: this is run in parallel to the main program
```

Comment:
It looks like, in the current form, ripgrep isn't actually run in parallel - it's blocking until all input lines are picked up. Are there plans to make this run in parallel?

For it to run in parallel - we might need a slightly different API. We might want a callback or a mechanism for it to stream results, and a callback for when it is complete. Then, as we get results, we can plumb them into the menu and apply the fuzzy matching (this is sort of how Oni1 worked)

Reply:
@bryphe the [documentation](https://caml.inria.fr/pub/docs/manual-ocaml/libref/Unix.html) for `open_process_in` in the Unix module states that it carries out the request in parallel

> val open_process_in : string -> in_channel

> High-level pipe and process management. This function runs the given command in parallel with the program. The standard output of the command is redirected to a pipe, which can be read via the returned input channel. The command is interpreted by the shell /bin/sh (or cmd.exe on Windows), cf. system.

My understanding was that getting the stream was async, but processing it as I do there is sync, but we do nothing in the processing other than pack it into a list

Reply:
So `open_process_in` does indeed open the process in parallel... but we're _waiting_ on the lines to come in - so the operation as a whole ends up being synchronous. 

This means there is lot of delay when I run `<C-p>` in the Oni2 repo - it's picking up a lot of files and waiting for them all to come in.

Ideally, pressing `<C-p>` should be instant (even if we're still streaming in results in the background), as well as typing to refine the search. 

We can follow up on this in a separate PR, though, if need be.

Reply:
I've switched to use `spawn` and emit an `Update` action with the new content whenever stuff comes through updating the menu content incrementally